### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by willy836**
**Milestones: Set up associations between classes and objects**

- [x] Create class Classroom and setup associations between Classroom and Student
- [x] Create classes Book and Rental and setup associations between Person, Book and Rental

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
-require './nameable'
-require './capitalize_decorator'
-require './trimmer_decorator'
+require_relative './nameable'
+require_relative './capitalize_decorator'
+require_relative './trimmer_decorator'
 
 class Person < Nameable
   attr_accessor :name, :age
```
Comment (julie-ify) [inline person.rb]: - [ ] Kindly add the getter and setter for `:rentals` as well.

File: person.rb [hunk 2]
```diff
@@ -27,13 +28,9 @@ def can_use_services?
...
-p person.correct_name
-person = Person.new(22, 'maximilianus')
-capitalized_person = CapitalizeDecorator.new(person)
-p capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-p capitalized_trimmed_person.correct_name
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (julie-ify) [inline person.rb]: - [ ] Your `add_rental` is somewhat redundant. For us to create a rental we need a person and the rental constructor automatically adds itself to the person's `rentals` array on creation. Now if we have created a rental already for a person. we don't really need to add it again here.

  The purpose of this method is for allowing the person object create a rental given only a date and book. Kindly refactor this method.

  ```
  def add_rental(date, book)
    Rental.new(date, book, self)
  end
  ```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    Rental.new(person, date, self)
+  end
```
Comment (julie-ify) [inline book.rb]: - [ ] Kindly rearrange your method here to match the order of parameters in the rental constructor.

  ```
  def add_rental(date, person)
    Rental.new(date, self, person)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @willy836,

Good job so far!

## Highlights!
- Use of getters and setters ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @willy836 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linters checks passed ✔️ 
- Set up associations between classes properly ✔️ 
- Pull request summary is descriptive ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**